build-support/bin/generate_docs.py

```diff
@@ -0,0 +1,301 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import html
+import json
+import logging
+import os
+import pkgutil
+import sys
+from pathlib import Path, PosixPath
+from typing import Dict, cast
+
+import pystache
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class ReferenceGenerator:
+    """Generates and uploads the Pants reference documentation.
+
+    To run use:
+
+    ./pants \
+      --backend-packages="-['internal_backend.rules_for_testing','internal_backend.utilities']" \
```

Comment:
An issue with the current site is that we don't activate all backends, specifically `pants.backend.python.lint.{bandit,pylint}` and `pants.backend.codegen.protobuf.python`. For the first two, you can permanently set `skip = true` in `[bandit]` and `[pylint]` to avoid us actually using them.

Reply:
Hmm, or can just add them in this cmd line for now.  If this gets cumbersome we can create a docgen-only config file.

Reply:
I think it might be good to still have them activated permanently. It's a very minimal form of dogfooding.